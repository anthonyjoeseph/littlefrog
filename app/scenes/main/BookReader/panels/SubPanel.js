// app/scenes/main/BookReader/panels/SubPanel.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Animated
} from 'react-native';

class SubPanel extends Component{
  constructor(props){
    super(props);

    this.state = {
      translateAnim: new Animated.Value(0),
      ownWidth: 1000, //initialize at a high value to avoid drawing before width is correctly updated
      isVisible: false,
      languages: this.languages
    }
  }

  slide(){
    var newFinalValue
    if(this.state.isVisible){
      newFinalValue = 0;
      this.state.isVisible = false;
    }else{
      newFinalValue = 1;
      this.state.isVisible = true;
    }
    Animated.timing(
      this.state.translateAnim,
      {
        toValue: newFinalValue,
        duration: 1000
      }
    ).start();
  }

  render(){
      return (
        <Animated.View
          style={
            [
              this.props.style,
              {
                transform:
                  [{
                    translateX: this.state.translateAnim.interpolate({
                      inputRange: [0, 1],
                      outputRange: [-this.state.ownWidth, 0]
                    })
                  }]
              }
            ]
          }
          onLayout={function(event){
            this.setState({ownWidth: event.nativeEvent.layout.width});
          }.bind(this)}>
          {this.props.children}
        </Animated.View>
      );
  }
}

export default SubPanel;
