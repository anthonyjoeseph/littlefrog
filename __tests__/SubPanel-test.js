// __tests__/SubPanel-test.js
import {View, Text} from 'react-native';
import React from 'react';
import SubPanel from '../app/scenes/main/BookReader/panels/SubPanel';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <SubPanel
      style={{
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'stretch'
        }}>
        <View>
          <Text>Testing 123</Text>
        </View>
    </SubPanel>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
