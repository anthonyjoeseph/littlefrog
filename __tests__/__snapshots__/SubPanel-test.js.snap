exports[`test renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "stretch",
      "flexDirection": "row",
      "justifyContent": "center",
      "transform": Array [
        Object {
          "translateX": -1000,
        },
      ],
    }
  }>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Testing 123
    </Text>
  </View>
</View>
`;
